### PR TITLE
Add com.rq3.Reaction

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/ReactionQuake3/reaction/pull/14
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/ReactionQuake3/reaction/pull/11 https://github.com/ReactionQuake3/reaction/pull/14

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build com.rq3.Reaction

File: com.rq3.Reaction.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    build-commands:
+      - make DEFAULT_BASEDIR=${FLATPAK_DEST}/share/Reaction -j${FLATPAK_BUILDER_N_JOBS}
+      - mkdir -p ${FLATPAK_DEST}/bin/
+      - mkdir -p ${FLATPAK_DEST}/share/Reaction/
+      - make copyfiles COPYBINDIR=${FLATPAK_DEST}/bin/ COPYDIR=${FLATPAK_DEST}/share/Reaction/ NO_STRIP=1
+      - mv ${FLATPAK_DEST}/bin/Reactionded.* ${FLATPAK_DEST}/bin/Reactionded
+      - mv ${FLATPAK_DEST}/bin/Reaction.* ${FLATPAK_DEST}/bin/Reaction
+      - mv ${FLATPAK_DEST}/bin/*.so ${FLATPAK_DEST}/share/Reaction/
+      - install -Dm644 logo.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm644 reaction.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
```
Comment (hfiguiere) [inline com.rq3.Reaction.yaml:29]: why aren't these two name properly from the get go since you submitted the patch?

File: com.rq3.Reaction.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: git
+        url: https://github.com/ReactionQuake3/reaction.git
+        tag: d214682124851c5fb4a1f7f254305ccfcec7bffc
+      - type: patch
+        path: resync.patch
```
Comment (hfiguiere) [inline com.rq3.Reaction.yaml:36]: a 5000+ lines patch, that's a bit excessive and divergent from upstream.

File: com.rq3.Reaction.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        use-git: true
+
+  - name: boomstick
+    buildsystem: simple
+    build-commands:
+      - rm -f *.x86_64 *.so
+      - cp -ar Boomstick ${FLATPAK_DEST}/share/Reaction
+    sources:
+      - type: archive
+        url: https://download.rq3.com/Reaction-1.0-Linux-x86_64.tar.gz
```
Comment (hfiguiere) [inline com.rq3.Reaction.yaml:48]: how does that work on aarch64?

File: com.rq3.Reaction.yaml (hunk 2 above)
Comment (Mailaender) [reply com.rq3.Reaction.yaml:36]: The necessary shader compilation crash fix is sadly included all in one commit which completes an ioQuake3 engine update. Both the 2012 release and the latest included commit are broken. Believe me, I tried a lot until I got this working.

File: com.rq3.Reaction.yaml (hunk 3 above)
Comment (Mailaender) [reply com.rq3.Reaction.yaml:48]: Notice that I just delete all platform specific pre-built binaries. I am only interested in the mod files, which are only included in these compiled forms together with an outdated engine. I could also pick the i386 download.

File: com.rq3.Reaction.yaml (hunk 1 above)
Comment (Mailaender) [reply com.rq3.Reaction.yaml]: I package software longer than Flathub exists and these naming rules are new and a bit counter-intuitive.

**Comment by Mailaender** [comment]
bot, build com.rq3.Reaction

**Pull request closed without merging**